src/bp-integrations/learndash/library/SyncGenerator.php

```diff
@@ -833,16 +833,27 @@ public function updateLearndashGroup( $ld_group_id, $groupId ) {
 		$bpGroup = groups_get_group( $groupId );
 
 		if ( ! empty( $ld_group_id ) ) {
-			wp_update_post(
-				array(
-					'ID'           => $ld_group_id,
-					'post_title'   => $bpGroup->name,
-					'post_author'  => $bpGroup->creator_id,
-					'post_content' => $bpGroup->description,
-					'post_status'  => 'publish',
-					'post_type'    => learndash_get_post_type_slug( 'group' ),
-				)
-			);
+			$ldGroup = get_post( $ld_group_id );
+			$args    = array();
+
+			if ( $bpGroup->name !== $ldGroup->post_title ) {
+				$args['post_title'] = $bpGroup->name;
+			}
+
+			if ( 'publish' !== $ldGroup->post_status ) {
+				$args['post_status'] = 'publish';
+			}
```

Comment:
@surajkrsingh Please remove this, as if someone updates the bb group and the learndash group is on draft at that time, it will automatically published, but it should not be as draft or trash will be intentionally by the user.